### PR TITLE
Release script: Update dependents count on release

### PR DESCRIPTION
**Pull request by fisker**
Update dependents count on release.

Unfortunately, I can't find any api for these two numbers.

@lipis Like it?

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: website/pages/en/index.js [hunk 1]
```diff
@@ -415,7 +415,8 @@ const UsersSection = ({ language }) => {
             </a>
             <div style={{ marginLeft: ".7em", width: "260px" }}>
               <p>
-                More than <strong>3000</strong> tools and integrations on npm
+                More than <strong data-name="dependent-npm">5.21 k</strong>{" "}
```
Comment (thorn0) [inline website/pages/en/index.js:418]: So `data-placeholder` or `data-name`? Also "More than 5000" would look less strange than "More than 5.21 k", IMHO.

Comment (fisker) [reply website/pages/en/index.js:418]: I use `data-placeholder` first... I don't know why I choose `data-name` when commit. 😄 

`human-format` format it this way

Comment (thorn0) [reply website/pages/en/index.js:418]: 5.21 k looks like a file size :)

Comment (fisker) [reply website/pages/en/index.js]: [human-number](https://npm.runkit.com/human-number) `5210` -> `5.21K` `1639226` -> `1.639226M` ...

Comment (fisker) [reply website/pages/en/index.js]: [hrn](https://runkit.com/embed/2lw28ebcp84x)  `["5 K", "1 M"]`

File: scripts/release/package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "dependencies": {
     "chalk": "2.4.1",
     "execa": "0.10.0",
+    "got": "11.1.3",
     "minimist": "1.2.3",
     "node-fetch": "2.1.2",
```
Comment (thorn0) [inline scripts/release/package.json:8]: We have `node-fetch` here already...

**Comment by thorn0** [comment]
Please wrap it in a `try ... catch` for the release script not to fail if something goes wrong with fetching that data.

**Comment by fisker** [comment]
You mean silence fail? [logPromise function](https://github.com/prettier/prettier/blob/master/scripts/release/utils.js#L22) already handle errors.

**Comment by thorn0** [comment]
Not silent. Print some info and proceed with the release. An extra check that the numbers look valid would be good to have too, so that we don't accidentally end up with "More than NaN tools..."

**Comment by fisker** [comment]
Success

```text
Fetching npm dependents count............................................. DONE
Fetching github dependents count.......................................... DONE
```

Network issue

```text
Fetching npm dependents count............................................. FAIL
request to https://----www.npmjs.com/package/prettier failed, reason: getaddrinfo EAI_AGAIN ----www.npmjs.com
(node:14552) UnhandledPromiseRejectionWarning: FetchError: request to https://----www.npmjs.com/package/prettier failed, reason: getaddrinfo EAI_AGAIN ----www.npmjs.com
    at ClientRequest.<anonymous> (<CWD>\prettier\scripts\release\node_modules\node-fetch\lib\index.js:1393:11)
    at ClientRequest.emit (events.js:310:20)
    at TLSSocket.socketErrorListener (_http_client.js:426:9)
    at TLSSocket.emit (events.js:310:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

Regex match error


```text
Fetching npm dependents count............................................. DONE
Cannot read property '1' of null
(node:16512) UnhandledPromiseRejectionWarning: TypeError: Cannot read property '1' of null
    at update (<CWD>\prettier\scripts\release\steps\update-dependents-count.js:15:47)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async <CWD>\prettier\scripts\release\steps\update-dependents-count.js:67:5
```

NaN

```text
Fetching npm dependents count............................................. DONE
Invalid data from https://www.npmjs.com/package/prettier
(node:17340) UnhandledPromiseRejectionWarning: TypeError: Invalid data from https://www.npmjs.com/package/prettier
    at update (<CWD>\prettier\scripts\release\steps\update-dependents-count.js:18:11)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async <CWD>\prettier\scripts\release\steps\update-dependents-count.js:67:5
```

IO

```text
Fetching npm dependents count............................................. DONE
Fetching github dependents count.......................................... DONE
ENOENT: no such file or directory, open 'website/pages/en/xindex.js'
(node:5772) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'website/pages/en/xindex.js'
    at Object.openSync (fs.js:458:3)
    at Object.readFileSync (fs.js:360:35)
	...
```

File: scripts/release/steps/update-dependents-count.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+  return Math.floor(value / 1e5) / 10 + "m";
+}
+
+module.exports = async function () {
+  try {
+    await update();
+  } catch (error) {
+    console.log(chalk.red.bold(error.message));
+    throw error;
```
Comment (thorn0) [inline scripts/release/steps/update-dependents-count.js:70]: Don't rethrow the error. This step isn't critical for the release. No need to stop the script.

File: scripts/release/release.js [hunk 4]
```diff
@@ -35,6 +35,7 @@ async function run() {
     require("./steps/install-dependencies"),
     require("./steps/run-tests"),
     require("./steps/update-version"),
+    require("./steps/update-dependents-count"),
```
Comment (thorn0) [inline scripts/release/release.js:38]: Let's move it to the end, between `bump-prettier` and `post-publish-steps`.

File: scripts/release/steps/update-dependents-count.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    return Math.floor(value / 1e2) / 10 + "k";
+  }
+  return Math.floor(value / 1e5) / 10 + "m";
+}
+
+module.exports = async function () {
+  try {
+    await update();
+  } catch (error) {
+    console.log(chalk.red.bold(error.message));
```
Comment (thorn0) [inline scripts/release/steps/update-dependents-count.js:69]:
```suggestion
    console.log(chalk.red.bold("Something went wrong with updating the counts, but don't worry, this step is optional. See the error below:"));
    console.log(error.message));
```

File: scripts/release/steps/update-dependents-count.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  );
+}
+
+function formatNumber(value) {
+  if (value < 1e4) {
+    return String(value).slice(0, 1) + "0".repeat(String(value).length - 1);
+  } else if (value < 1e6) {
+    return Math.floor(value / 1e2) / 10 + "k";
+  }
+  return Math.floor(value / 1e5) / 10 + "m";
```
Comment (thorn0) [inline scripts/release/steps/update-dependents-count.js:62]: Where does GitHub use the lowercase m? I thought these abbreviations were based on the SI prefixes (kilo = k, mega = M, etc.).

**Pull request merged**